src/cryptsetup/cryptsetup.c

```diff
@@ -621,7 +891,7 @@ int main(int argc, char *argv[]) {
                     !streq(argv[4], "-") &&
                     !streq(argv[4], "none")) {
 
-                        if (!path_is_absolute(argv[4]))
+                        if (!path_is_absolute(argv[4]) && !startswith(argv[4], "pkcs11:"))
                                 log_error("Password file path '%s' is not absolute. Ignoring.", argv[4]);
                         else
                                 key_file = argv[4];
```

Comment:
is there any circumstance where key_file might be changed and no longer start with pkcs11: when you go to look for it later? Seems ok though

It might be cool to set the stage for further expansion/extensions, switch statement? Break out isome functionality? Idk... really excited about this pr though, need good support for hsms etc